angular-ngrx-scss/src/app/shared/components/repo-controls/repo-filter-dropdown/repo-filter-dropdown.component.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Input,
+  Output,
+} from '@angular/core';
+import { trigger, transition, style, animate } from '@angular/animations';
+
+export interface FilterOption {
+  label: string;
+  value: string;
+}
+
+@Component({
+  selector: 'app-repos-filter-dropdown',
+  templateUrl: './repo-filter-dropdown.component.html',
+  styleUrls: ['./repo-filter-dropdown.component.scss'],
+  animations: [
+    trigger('openClose', [
+      transition(':enter', [
+        style({ opacity: 0, transform: 'scale(0.95)' }),
+        animate('100ms ease-out', style({ opacity: 1, transform: 'scale(1)' })),
+      ]),
+      transition(':leave', [
+        animate('75ms', style({ opacity: 0, transform: 'scale(0.95)' })),
+      ]),
+    ]),
+  ],
+})
```

Comment:
it seems this component does not mutate itself and gets all its data from a parent? do we mean to change the change detection strategy?